cdap-ui/app/cdap/components/HttpExecutor/store/HttpExecutorStore.js

```diff
@@ -62,10 +63,21 @@ const setResponse = (state, action) => {
     timestamp: new Date().toLocaleString(),
   };
 
-  const newRequestLog = requestLog.push(newCall);
+  // Update the component view in real-time, since we cannot listen to local storage's change
+  // Since the new request call is the latest out of all the request histories, insert at 0th index
+  const timestamp = new Date(newCall.timestamp);
+  const dateID = getDateID(timestamp);
+  const requestsGroup = getRequestsByDate(requestLog, dateID);
+  const newRequestLog = requestLog.set(dateID, requestsGroup.insert(0, newCall));
 
-  // saving of the request to the localStorage
-  localStorage.setItem(REQUEST_HISTORY, JSON.stringify(newRequestLog));
+  // Saving request histories to the localStorage
+  let storedLogs = List([]);
+  Array.from(newRequestLog.valueSeq()).forEach((requests) => {
```

Comment:
isn't `valueSeq()` already returns a List? do you still need the `Array.from`?

Reply:
valueSeq returns iterator